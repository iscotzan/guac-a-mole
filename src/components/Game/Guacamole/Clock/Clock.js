import React from "react";
import { Circle } from "rc-progress";

const Clock = props => {
  const countText =
    props.currentCount >= 10 ? props.currentCount : "0" + props.currentCount;
  return (
    <div className="timerContainer">
      <div className={"countText"}>{countText}</div>
      <Circle
        percent={props.percentTimeLeft}
        strokeWidth="6"
        strokeLinecap="round"
        className={"timeCircle"}
      />
      <div />
    </div>
  );
};

export default Clock;
