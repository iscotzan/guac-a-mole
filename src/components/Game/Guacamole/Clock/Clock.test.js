import React from "react";
import Clock from "./Clock";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<Clock percentTimeLeft={100} currentCount={60} />);
describe("<Clock />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
