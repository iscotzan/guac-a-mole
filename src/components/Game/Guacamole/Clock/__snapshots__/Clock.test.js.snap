// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> renders without crashing 1`] = `
<div
  className="timerContainer"
>
  <div
    className="countText"
  >
    60
  </div>
  <Progress
    className="timeCircle"
    gapPosition="top"
    percent={100}
    prefixCls="rc-progress"
    strokeColor="#2db7f5"
    strokeLinecap="round"
    strokeWidth="6"
    style={Object {}}
    trailColor="#D9D9D9"
    trailWidth={1}
  />
  <div />
</div>
`;
